sync/sync_test.go

```diff
@@ -349,6 +349,36 @@ func TestSync_ShouldRegisterMissingRegistrationInMultiregistrationScenario(t *te
 	assert.Len(t, services, 2)
 }
 
+/*
+This may happen if an application configuration is changed and then the deployment this triggered is rolled back.
+After rollback, the new configuration still stays with the application and there's no way to access the original configuration
```

Comment:
Then, it should be explicitly mentioned that this is special case for deployment cancelation or add example request that trigger this state 

    DELETE /v2/deployments/177b7556-1287-4e09-8432-3d862981a987?force=true